src/Models/Viewpoint.php

```diff
@@ -155,6 +230,23 @@ public function findActionGroupById(string $actionGroupId)
         return null;
     }
 
+    /**
+     * Checks if the viewpoint has a certain action group.
+     * @param string $actionGroupId
+     * @return bool
+     */
+    public function hasActionGroup(string $actionGroupId): bool
```

Comment:
Do we really need this and findActionGroupById? I dont think so.

And if we do, let's just use findActionGroupById() !=== null as the implementation.

Reply:
I suppose renaming the find-Method to a getMethod would be better. Having both makes the code a bit easier to read:

```php
if ($v->hasActionGroup($actionGroupId)) {
  $v->getActionGroup($actionGroupId)->addAction($action);
} else {
  $v->addActionGroup(
    (new ActionGroup($actionGroupId))
    ->addAction($action)
  );
}
```

vs

```php
$group = $v->getActionGroup($actionGroupId);
if ($group) {
  $group->addAction($action);
} else {
  $v->addActionGroup(
    (new ActionGroup($actionGroupId))
    ->addAction($action)
  );
}
```

Reply:
Im indifferent to whether it exists. Go with your gut here. But dont write two implementations :)